janus_server/src/trace.rs

```diff
@@ -109,6 +139,84 @@ pub fn install_trace_subscriber(config: &TraceConfiguration) -> Result<(), Error
         );
     }
 
+    if config.otel_jaeger && config.otel_otlp.enabled {
+        return Err(Error::Other(
+            "bad configuration (both Jaeger and OTLP OpenTelemetry layers cannot be enabled at the same time)"
+        ));
+    }
+
+    #[cfg(feature = "jaeger")]
+    if config.otel_jaeger {
+        let tracer = opentelemetry_jaeger::new_pipeline()
+            .with_service_name("janus_server")
+            .install_batch(opentelemetry::runtime::Tokio)?;
+        let telemetry = tracing_opentelemetry::layer().with_tracer(tracer);
+        layers.push(telemetry.boxed());
+    }
+
+    #[cfg(not(feature = "jaeger"))]
+    if config.otel_jaeger {
+        eprintln!(
```

Comment:
IMO, trying to enable features not compiled in should result in a failure, rather than a warning. A warning may not be noticed & means we'll just keep rolling with a difference between what we think we're doing & what we're actually doing. A failure is a lot harder to ignore.

[same comment for the warning around L213]